src/vm/lib/metadata/agent.js

```diff
@@ -1282,11 +1441,32 @@ MetadataAgent.prototype.makeMetadataHandler = function (zone, socket) {
 
         function addMetadata(cb) {
             var filename;
+            var start_add_metadata = (new Date()).getTime();
+
+            function _callCbAndLogTimer(err) {
+                var elapsed = (new Date()).getTime() - start_add_metadata;
```

Comment:
Maybe here using a monotonic clock instead of wall-clock time could prevent some inconsistent results (false positives and false negatives)?

In general I think we could use monotonic clock time everywhere in this PR where we currently use wall-clock time, but I'll let you make the call with regards to when and where to fix that.